pandas/io/pytables.py

```diff
@@ -1877,7 +1879,7 @@ def set_atom_string(self, block, block_items, existing_col, min_itemsize,
         self.itemsize = itemsize
         self.kind = 'string'
         self.typ = self.get_atom_string(block, itemsize)
-        self.set_data(self.convert_string_data(data, itemsize, encoding))
+        self.set_data(data_converted.astype('|S%d' % itemsize))
```

Comment:
We still need the `astype` though I think, since we might have changed `itemsize` be what the user passed in for `min_itemsize`:  `itemsize = max(min_itemsize or 0, itemsize)`.